test/gtest/common/test_helpers.h

```diff
@@ -651,6 +651,40 @@ class handle {
     ArgT           m_dtor_arg;
 };
 
+/* simplified version of std::auto_ptr which was deprecated in newer stdc++
+ * versions in favor of unique_ptr */
+template <typename T>
+class auto_ptr {
+public:
+    auto_ptr() : m_ptr(NULL) {
+    }
+
+    auto_ptr(T* ptr) {
```

Comment:
```auto_ptr(T* ptr): m_ptr(NULL) {```

Reply:
m_ptr would be initialized by "reset" below